### PR TITLE
[MU4] Straight flags now spaced better

### PR DESCRIPTION
**Pull request by asattely**
Traditional flags on notes are reduced outside the staff such as to make sure the hook ends up at a certain vertical point, but this is meaningless for straight flags. The shortening that straight-flagged notes do now follows the same rules as beams:
image

### CODE REVIEW

**Review by oktophonie — APPROVED** [review]

File: src/engraving/libmscore/chord.cpp [hunk 1]
```diff
@@ -1830,6 +1836,15 @@ void Chord::layoutStem()
 {
     // we should calculate default stem length for this chord even if it doesn't have a stem
     // because this length is used for tremolos or other things that attach to where the stem WOULD be
```
Comment (cbjeukendrup) [inline src/engraving/libmscore/chord.cpp:1838]: Shouldn't this comment stay together with the line `_defaultStemLength = calcDefaultStemLength();`?

**Review by cbjeukendrup — APPROVED** [review]
It feels a little bit scary to layout the hook potentially before the stem has been created. This means we can't use the stem in Hook::layout. However, as long as we are aware of that, it's not per se a problem, and I currently don't see any reason why we should depend on the stem in Hook::layout. To be fair, it's the other way around at the moment. 
For the rest a cosmetic comment only!

**Comment by RomanPudashkin** [comment]
@asattely please rebase it

**Review by RomanPudashkin — APPROVED** [review]

**Comment by RomanPudashkin** [comment]
@asattely sorry, but please rebase it again...

File: src/engraving/libmscore/chord.cpp (hunk 1 above)
Comment (asattely) [reply src/engraving/libmscore/chord.cpp]: yep, good catch, fixed

**Comment by asattely** [comment]
Rebased

**Pull request merged**